dace/frontend/fortran/ast_transforms.py

```diff
@@ -130,17 +332,79 @@ def visit_Array_Subscript_Node(self, node: ast_internal_classes.Array_Subscript_
         for i in node.indices:
             self.visit(i)
 
+    def visit_Data_Ref_Node(self, node: ast_internal_classes.Data_Ref_Node):
+        if isinstance(node.parent_ref, ast_internal_classes.Name_Node):
+            self.nodes.append(node.parent_ref)
+        elif isinstance(node.parent_ref, ast_internal_classes.Array_Subscript_Node):
+            self.nodes.append(node.parent_ref.name)
+        if isinstance(node.parent_ref, ast_internal_classes.Array_Subscript_Node):
+            for i in node.parent_ref.indices:
+                self.visit(i)
+        if isinstance(node.part_ref, ast_internal_classes.Array_Subscript_Node):
+            for i in node.part_ref.indices:
+                self.visit(i)
+        elif isinstance(node.part_ref, ast_internal_classes.Data_Ref_Node):
+            self.visit_Blunt_Data_Ref_Node(node.part_ref)
+
+    def visit_Blunt_Data_Ref_Node(self, node: ast_internal_classes.Data_Ref_Node):
+        if isinstance(node.parent_ref, ast_internal_classes.Array_Subscript_Node):
+            for i in node.parent_ref.indices:
+                self.visit(i)
+        if isinstance(node.part_ref, ast_internal_classes.Array_Subscript_Node):
+            for i in node.part_ref.indices:
+                self.visit(i)
+        elif isinstance(node.part_ref, ast_internal_classes.Data_Ref_Node):
+            self.visit_Blunt_Data_Ref_Node(node.part_ref)
+
     def visit_BinOp_Node(self, node: ast_internal_classes.BinOp_Node):
         if node.op == "=":
             if isinstance(node.lval, ast_internal_classes.Name_Node):
                 pass
             elif isinstance(node.lval, ast_internal_classes.Array_Subscript_Node):
                 for i in node.lval.indices:
                     self.visit(i)
+            elif isinstance(node.lval, ast_internal_classes.Data_Ref_Node):
+                # if isinstance(node.lval.parent_ref, ast_internal_classes.Name_Node):
```

Comment:
If these commented out lines still represent something that needs to be fixed/ investigated, please add a TODO / TODO(later), or remove them otherwise.